### PR TITLE
Add fake cookie support to FakeWin

### PR DESCRIPTION
**Pull request by jridgewell**
Needed for me to complete #8108.

Do we have tests for our testing infrastructure? If not, test with:

```js
document.cookie = 'test=value';
document.cookie; // => 'test=value'

document.cookie = 'test2=value2';
document.cookie; // => 'test=value;test2=value2'

document.cookie = 'test=value3';
document.cookie; // => 'test=value3;test2=value2'

document.cookie = 'test2=value4';
document.cookie; // => 'test=value3;test2=value4'

document.cookie = 'test2=value5;expires=' new Date(0).toUTCString();
document.cookie; // => 'test=value3'

document.cookie = 'test=value6';
document.cookie; // => 'test=value6'

document.cookie = 'test3=value7'
document.cookie; // => 'test=value3;test3=value7'

document.cookie = 'test=value8;expires=' new Date(0).toUTCString();
document.cookie; // => 'test3=value7'

document.cookie = 'test3=value8;expires=' new Date(0).toUTCString();
document.cookie; // => ''

### CODE REVIEW

**Comment by erwinmombay** [comment]
who owns testing/* ?

**Comment by jridgewell** [comment]
@dvoytenko does.

```
git ls-tree -r -z --name-only HEAD -- testing | xargs -0 -n1 git blame \
--line-porcelain HEAD |grep  "^author "|sort|uniq -c|sort -nr
15073 author Dima Voytenko
 441 author Malte Ubl
  95 author Hongfei Ding
  72 author Ali Ghassemi
  57 author Justin Ridgewell
  48 author Yuxuan Zhou
  34 author Mohammad Khatib
  21 author William Chou
  13 author Erwin Mombay
   9 author Terran Lane
   6 author erwin mombay
   5 author Yuxi Chen
   4 author Marcin Orczyk
   3 author Sriram Krishnan
   1 author Meggin Kearney
```

File: testing/fake-dom.js [hunk 1]
```diff
@@ -98,6 +98,35 @@ export class FakeWindow {
...
+        const cookie = value.match(/^([^=]*)=([^;]*)/);
+        const expiresMatch = value.match(/expires=([^;]*)(;|$)/);
+        const expires = expiresMatch ? Date.parse(expiresMatch[1]) : Infinity;
+        let i = 0;
+        for (; i < this.cookie_.length; i += 2) {
+          if (this.cookie_[i] == cookie[1]) {
+            break;
+          }
+        }
+        if (Date.now() >= expires) {
```
Comment (dvoytenko) [inline testing/fake-dom.js]: Shouldn't expiration be checked in the `get`?

**Comment by dvoytenko** [comment]
@jridgewell LGTM with one question on expiry and getter path.

File: testing/fake-dom.js (hunk 1 above)
Comment (jridgewell) [inline testing/fake-dom.js]: Ideally, but only real use of expires I'm aware of is to delete the cookie immediately. So this worked well.

**Pull request merged**